### DISCUSSION TITLE
code annotation and code linking

### DISCUSSION THREAD

**Discussion by PierreSolomon** [quarto-dev/quarto-cli#6187] [Feature Requests]
### Description

Hello,

I just tried to render a document using both code annotation and code linking with the downlit packages.

After the rendering I get the message next:
```
Message d'avis :
Dans postprocess(params$input, params$format, params$output, params$preserve) :
  Since code annotations are present, code-linking has been disabled
```
Those features would be interesting to use together as they both permit to better explain the code. Is there a possibility that in the future, they can be used together in the same document and code chunk? 

Thank you in advance for the answer

**Comment by dragonstyle** [comment]
Currently, code annotation has some dependency upon the structure of the HTML output within the code cell (in order to do things like position the annotation numbers). Downlit modifies the structure enough that we haven't found a way to keep the annotation functionality working with it - it is something I would like to support but I'd say right now I don't think this is something that will happen anytime soon...

**Reply by mcanouil** [reply]
Should we highlight that in the reference pages as another limitation?

https://quarto.org/docs/reference/formats/html.html#code

![image](https://github.com/quarto-dev/quarto-cli/assets/8896044/2283c072-9bed-4cda-8a87-87280f6da1d3)

**Reply by dragonstyle** [reply]
Sure that would make sense!

**Reply by mcanouil** [reply]
I opened an issue to track this and will likely take care of it.

- #6199

**Comment by PierreSolomon** [comment]
Thank you for the quick answer. I guess I will stick with the use of comments to annotate my code.

**Comment by mcanouil** [comment]
Closing this for now in favour of #6199 for improved documentation.

Since `downlit` will likely require heavy and not only for `code-annotation` a dedicated issue will be opened in due time.
